### PR TITLE
fix the hard coded feature gate

### PR DESCRIPTION
**Pull request by wangchen615**
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/node/pod_resize.go [hunk 1]
```diff
@@ -276,10 +272,6 @@ func doPodResizeTest() {
 	var podClient *framework.PodClient
 	var ns string
 
-	if !utilfeature.DefaultFeatureGate.Enabled(InPlacePodVerticalScalingFeature) {
```
Comment (wangchen615) [inline test/e2e/node/pod_resize.go]: @vinaykul @ehashman, please correct me if I am wrong. So I made the e2e testing changes according to our previous discussion.
 `so we don't call SIGDescribe anywhere in the e2e tests that are defined.  this is mandatory, they have to be tagged by owner SIG. furthermore we don't have any selectors defined that will indicate whether or not the tests should run. typically we do this with a [Feature:Foo] tag on the test name. then we modify the GCP alpha job, and run that, we don't necessarily need to make a special node one.`

Comment (wangchen615) [reply test/e2e/node/pod_resize.go]: And the PR we have to modify the GCP alpha job is here [test-infra#23177](https://github.com/kubernetes/test-infra/pull/23177)

Comment (wangchen615) [reply test/e2e/node/pod_resize.go]: @ehashman, could you help review both?

**Comment by vinaykul** [comment]
I had to redo the code and split out the API, implementation, and generated files into different commits as it was before the squash few months ago. Instead of taking up your time again just to rebase the code, I pulled down your patch and applied it, and I've mentioned you as co-author.
Thanks a lot for working on this.

**Comment by wangchen615** [comment]
> I had to redo the code and split out the API, implementation, and generated files into different commits as it was before the squash few months ago. Instead of taking up your time again just to rebase the code, I pulled down your patch and applied it, and I've mentioned you as co-author. Thanks a lot for working on this.

Thank you @vinaykul for all these works. Feel free to close it when you get the test.

**Comment by wangchen615** [comment]
@vinaykul , I reopened this so @ehashman can take a final look.

**Pull request closed without merging**